### PR TITLE
Rework how disambiguation is handled

### PR DESCRIPTION
**Pull request by SimonBrandner**
This adds a `disambiguate` property to the `RoomMember`. This is useful when a client wants to handle more things on its own. For example, a client may wish to format the `displayName` differently than the `userId` and therefore has to retrieve the `displayName` and `userName` separately - it can't rely on `js-sdk`'s calculation of display names.

### CODE REVIEW

File: src/models/room-member.js [hunk 1]
```diff
@@ -45,6 +45,7 @@ import * as utils from "../utils";
  * @prop {string} membership The membership state for this room member e.g. 'join'.
  * @prop {Object} events The events describing this RoomMember.
  * @prop {MatrixEvent} events.member The m.room.member event for this RoomMember.
+ * @prop {boolean} disambiguate True if the client show disambiguate display names
```
Comment (ShadowJonathan) [inline src/models/room-member.js:48]:
```suggestion
 * @prop {boolean} disambiguate True if the client should disambiguate display names
```

Comment (SimonBrandner) [reply src/models/room-member.js]: Thanks for catching that!

File: src/models/room-member.js [hunk 2]
```diff
@@ -45,6 +45,7 @@ import * as utils from "../utils";
  * @prop {string} membership The membership state for this room member e.g. 'join'.
  * @prop {Object} events The events describing this RoomMember.
  * @prop {MatrixEvent} events.member The m.room.member event for this RoomMember.
+ * @prop {boolean} disambiguate True if the client should disambiguate display names
```
Comment (turt2live) [inline src/models/room-member.js:48]: This is an indicator flag and not something that can be set external to the room member, so we should word it as such.
```suggestion
 * @prop {boolean} disambiguate True if the member's name is disambiguated.
```

**Review by turt2live — APPROVED** [review]
otherwise lgtm, thanks!

**Review by turt2live — APPROVED** [review]
looks good - we'll probably land this at the same time as the react-sdk, just in case.

**Comment by SimonBrandner** [comment]
New PR at: https://github.com/matrix-org/matrix-js-sdk/pull/1730

**Pull request closed without merging**